### PR TITLE
Decouple "mkdirp".

### PR DESCRIPTION
**Pull request by ysmood**
Make coffee completely independent.
As such a great project, coffee should keep as light as possible. Let's decouple the last one~

### CODE REVIEW

**Comment by ysmood** [comment]
@michaelficarra Will this PR be merged?

**Comment by michaelficarra** [comment]
I'd feel more comfortable if there were tests related to this.

**Comment by ysmood** [comment]
@michaelficarra Well, I'll add one.

**Comment by ysmood** [comment]
@michaelficarra A simple test has been added. How you test command line functions of coffee? I cannot find any.

File: test/mkdirp.coffee [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  process.argv = [
+    'node', 'coffee'
+    '-o', deepDir
+    '-c', __filename
+  ]
+  command.run()
+
+  compiledPath = path.join(
+    deepDir
+    path.basename(__filename, '.coffee') + '.js'
```
Comment (michaelficarra) [inline test/mkdirp.coffee]: Don't use + for string concatenation.

File: test/mkdirp.coffee [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  )
+
+  # Coffee's test suit doesn't support async task.
+  # Here some hack.
+  setTimeout ->
+    ok fs.existsSync(compiledPath)
```
Comment (michaelficarra) [inline test/mkdirp.coffee]: Just move the opening paren before the arrow and we don't have to do this gross line continuation.

**Comment by jashkenas** [comment]
I'm not sure that we should be adding tests that muck around with the file system, and rely on process.exit not being killed or interrupted early.

But otherwise, let's merge this! Wonderful to get rid of the only dependency.

File: test/mkdirp.coffee [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  , 1000
```
Comment (epidemian) [inline test/mkdirp.coffee]: I don't see why this setTimeout of 1000 ms is necessary. Can't the same assertion be run on the `'exit'` callback of the process?

Comment (ysmood) [inline test/mkdirp.coffee]: @epidemian Good idea, I slightly altered the test, now it's more comfortable.

**Comment by ysmood** [comment]
Removed the `setTimeout` in the test. Still very tricky though.

**Comment by ysmood** [comment]
@jashkenas Removed the `setTimeout` in the test. Still very tricky though.

**Comment by jashkenas** [comment]
I'm going to merge, and then tweak, or maybe kill the test.

**Pull request merged**